server/routes.go

```diff
@@ -255,10 +255,9 @@ func getAccessHandler(m KeyManager, principal knox.Principal, parameters map[str
 		return nil, errF(knox.InternalServerErrorCode, getErr.Error())
 	}
 
-	// Authorize
```

Comment:
There's a comment on line 244 that contradicts your comment: "The principal needs Read access." You should get rid of it.